modules/core/src/test/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorageSuite.scala

```diff
@@ -0,0 +1,65 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.effect.{IO, Resource}
+import cats.syntax.either._
+import cats.syntax.option._
+
+import org.tessellation.dag.dagSharedKryoRegistrar
+import org.tessellation.dag.snapshot.SnapshotOrdinal
+import org.tessellation.ext.crypto._
+import org.tessellation.infrastructure.snapshot.GlobalSnapshotStorage.InvalidGlobalSnapshotChain
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.sdk.kryo.sdkKryoRegistrar
+
+import monocle.syntax.all._
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object GlobalSnapshotStorageSuite extends MutableIOSuite with Checkers {
+
+  type Res = KryoSerializer[IO]
+
+  override def sharedResource: Resource[IO, GlobalSnapshotStorageSuite.Res] =
+    KryoSerializer.forAsync[IO](dagSharedKryoRegistrar ++ sdkKryoRegistrar)
+
+  test("should accept valid snapshot") { implicit kryo =>
+    for {
+      storage <- GlobalSnapshotStorage.make(genesis)
+      genesisHash <- genesis.hash.liftTo[IO]
+      nextSnapshot = genesis
+        .focus(_.ordinal)
+        .modify(SnapshotOrdinal.next.next)
+        .focus(_.lastSnapshotHash)
+        .replace(genesisHash)
+      _ <- storage.save(nextSnapshot)
+    } yield success
+  }
+
+  test("should not accept snapshot with invalid `lastSnapshotHash`") { implicit kryo =>
+    for {
+      storage <- GlobalSnapshotStorage.make(genesis)
+      nextSnapshot = genesis
+        .focus(_.ordinal)
+        .modify(SnapshotOrdinal.next.next)
+      maybeError <- storage
+        .save(nextSnapshot)
+        .map(_ => none[Throwable])
+        .handleError(_.some)
+    } yield verify(maybeError.exists(_.isInstanceOf[InvalidGlobalSnapshotChain]))
```

Comment:
I think that you can expect `failure` instead of mapping an error to option and checking the option itself 🤔

Reply:
Failure is not the same as expecting an error. Failure means that you unconditionally fail the test. I haven't found any syntax for expecting an error in Weaver.

Reply:
Ah, right. I misunderstood the documentation.